File: test-suite/src/test/java/io/micronaut/kafka/docs/consumer/sendto/QuantityListener.java
package io.micronaut.kafka.docs.consumer.sendto;

import io.micronaut.configuration.kafka.annotation.KafkaKey;
import io.micronaut.configuration.kafka.annotation.KafkaListener;
import io.micronaut.configuration.kafka.annotation.OffsetReset;
import io.micronaut.configuration.kafka.annotation.Topic;
import io.micronaut.context.annotation.Requires;
import org.slf4j.Logger;

import static org.slf4j.LoggerFactory.getLogger;

@Requires(property = "spec.name", value = "SendToProductListenerTest")
@KafkaListener(offsetReset = OffsetReset.EARLIEST)
public class QuantityListener {

    private static final Logger LOG = getLogger(QuantityListener.class);

    Integer quantity;

    @Topic("product-quantities")
    public void receive(@KafkaKey String brand, Integer quantity) {
        LOG.info("Got quantity - {} by {}", quantity, brand);
        this.quantity = quantity;
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/consumer/sendto/QuantityListener.kt
package io.micronaut.kafka.docs.consumer.sendto

import io.micronaut.configuration.kafka.annotation.KafkaKey
import io.micronaut.configuration.kafka.annotation.KafkaListener
import io.micronaut.configuration.kafka.annotation.OffsetReset
import io.micronaut.configuration.kafka.annotation.Topic
import io.micronaut.context.annotation.Requires
import io.micronaut.kafka.docs.consumer.batch.BookListener
import org.slf4j.LoggerFactory

@Requires(property = "spec.name", value = "SendToProductListenerTest")
@KafkaListener(groupId = "send-to-product-group", offsetReset = OffsetReset.EARLIEST)
class QuantityListener {

    companion object {
        private val LOG = LoggerFactory.getLogger(BookListener::class.java)
    }

    var quantity: Integer? = null

    @Topic("product-quantities")
    fun receive(@KafkaKey brand: String?, quantity: Integer) {
        LOG.info("Got Quantity - {} by {}", quantity, brand)
        this.quantity = quantity
    }
}
